From 1549bb56dfa9a4b960b74c269f97305f9e44ee3f Mon Sep 17 00:00:00 2001
From: wangwen <wangwen@marvell.com>
Date: Wed, 16 May 2018 16:55:54 +0800
Subject: [PATCH 0647/1051] dts: armada-37xx: reserve 16M memory for tee

These memory are used for TEE code, data, communication channel
and shared memory with Non-Trusted world.

For the TEE code and data section, it will be configured
unacceable to Non-Trusted World by memory controller.

The rest memories are either used for communication or shared
memory between Trusted-World and Non-trusted World, which is
TEE implementation defined and may be managed by the TEE driver.

Change-Id: I19536da77513e39a195d5fffeeeb0068c089f7e5
Signed-off-by: wangwen <wangwen@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index f622a2c5903c..d5a699b4a1ef 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -35,6 +35,11 @@
 			reg = <0 0x4000000 0 0x200000>;
 			no-map;
 		};
+
+		tee@4400000 {
+			reg = <0 0x4400000 0 0x1000000>;
+			no-map;
+		};
 	};
 
 	cpus {
-- 
2.17.1

